packages/web/src/pages/search-page-v2/SearchResults.tsx

```diff
@@ -0,0 +1,257 @@
+import { Text } from '@audius/harmony/src/components/text'
+import { css } from '@emotion/css'
+import { range } from 'lodash'
+import { useDispatch } from 'react-redux'
+
+import { make } from 'common/store/analytics/actions'
+import { CollectionCard } from 'components/collection'
+import Lineup from 'components/lineup/Lineup'
+import { LineupVariant } from 'components/lineup/types'
+import { UserCard } from 'components/user-card'
+import { useRouteMatch } from 'hooks/useRouteMatch'
+import { useSelector } from 'utils/reducer'
+import { SEARCH_CATEGORY_PAGE } from 'utils/route'
+
+const { getUserId } = accountSelectors
+
+const MAX_RESULTS = 100
+const MAX_PREVIEW_RESULTS = 5
+
+enum Category {
+  ALL = 'all',
+  PROFILES = 'profiles',
+  TRACKS = 'tracks',
+  PLAYLISTS = 'playlists',
+  ALBUMS = 'albums'
+}
+
+const messages = {
+  profiles: 'Profiles',
+  tracks: 'Tracks',
+  albums: 'Albums',
+  playlists: 'Playlists'
+}
+
+// const ResultsUserCard = ({ id }: { id: number }) => {
+//   const { status } = useGetUserById({ id })
+//   return <UserCard id={id} size='s' loading={status === Status.LOADING} />
+// }
+
+// const ResultsCollectionCard = ({ id }: { id: number }) => {
+//   const { status } = useGetPlaylistById({ playlistId: id })
+//   return <CollectionCard id={id} size='s' loading={status === Status.LOADING} />
+// }
+
+type SearchResultsProps = {
+  query: string
+}
+
+const cardGridStyles = {
+  display: 'grid',
+  gridTemplateColumns: 'repeat(auto-fill, 200px)',
+  justifyContent: 'space-between',
+  gap: 16
+}
+
+const getCurrentQueueItem = makeGetCurrent()
+const getTracksLineup = makeGetLineupMetadatas(getSearchTracksLineup)
+
+export const SearchResults = ({ query }: SearchResultsProps) => {
+  const containerRef = useRef<HTMLDivElement>(null)
+  const currentUserId = useSelector(getUserId)
+  const currentQueueItem = useSelector(getCurrentQueueItem)
+  const tracks = useSelector(getTracksLineup)
+  const playing = useSelector(getPlaying)
+  const buffering = useSelector(getBuffering)
+  const results = useGetSearchFull({ currentUserId, query })
+  const categoryMatch = useRouteMatch<{ query: string; category: string }>(
+    SEARCH_CATEGORY_PAGE
+  )
+
+  const dispatch = useDispatch()
+  useEffect(() => {
+    dispatch(fetchSearchPageResults(query, SearchKind.ALL, 50, 0))
+  }, [dispatch, query])
+
+  const [isLoading, setIsLoading] = useState(true)
+  const prevQuery = useRef('')
+  useEffect(() => {
+    if (results.data && isLoading && query !== prevQuery.current) {
+      prevQuery.current = query
+      setIsLoading(false)
+    } else if (!isLoading && query !== prevQuery.current) {
+      setIsLoading(true)
+    }
+  }, [results, query, isLoading])
+
+  const isCategoryVisible = useCallback(
+    (category: Category) =>
+      !categoryMatch ||
+      categoryMatch.category === Category.ALL ||
+      categoryMatch.category === category,
+    [categoryMatch]
+  )
+
+  const profileLimit =
```

Comment:
Cool, yeah this looks good for now like we discussed we can always split into multiple components if need be